src/openforms/appointments/api/views.py

```diff
@@ -388,6 +390,13 @@ def get_serializer_context(self):
         context.update({"submission": submission})
         return context
 
+    @transaction.atomic
+    def create(self, request: Request, *args, **kwargs):
+        # ensure any previous attempts are deleted before creating a new one
+        submission = self.extract_submission()
+        Appointment.objects.filter(submission=submission).delete()
```

Comment:
Can there be other, existing appointments we don't want to delete?

Reply:
No, it's a `OneToOneField`, which is also where the unique validation error is coming from.